### DISCUSSION TITLE
How to control your data usage

### DISCUSSION THREAD

**Discussion by enague** [mixpanel/docs-legacy#526] [Tips and Tricks]
Your data usage is likely to fluctuate as your product changes, you add new features, and your business expands. All Mixpanel plans have a cap on the number of events or MTUs that can be monitored. Once that limit is reached, you’ll either pay the additional usage as overages at an “à la carte” price, which is slightly more expensive than the data points included in your license, or your account will be temporarily blocked until the new billing cycle (month) begins.

**We don’t want you to pay overages or be blocked!**
If you detect you are close to hitting your plan's limit during the course of your billing cycle, you can always upgrade via [our Pricing Page](https://mixpanel.com/pricing/).

Billing surprises are never nice, so here are a few ideas on how to better control your data usage:
### 1. First, get familiar with what your plan includes

If you have “Owner” or “Billing Admin” permissions over your Mixpanel organization, you’ll be able to access the **"Plan Details & Billing"** section in your Organization Settings:

<p align="center">
<img alt="img" src="https://github.com/user-attachments/assets/f3f4f854-b52d-4f54-aaf2-7a767956e95e" />
</p>

In the “My Plan” tab, you’ll see if your plan is based on **events** or **Monthly Tracked Users** (MTUs), when your current subscription ends, and a real-time count of the current billing cycle's data usage - the example above is an event-based plan that includes 200K events per month.

You can also check the usage from past billing cycles and/or per project by accessing the "Data Usage" tab and clicking the "View Historical Usage" button:

<p align="center">
<img width="75%" height="75%" alt="2026_Data_Usage_tab" src="https://github.com/user-attachments/assets/8e2f0af8-2538-47ff-b23d-a0a079272efc" />
</p>

### 2. Keep an eye on your email!

Our Finance team has automated email notifications so "Owners" and "Billing Admins" get notified when they are about to hit the limit of their current plan. Remember to upgrade **before** the end of the current billing cycle so you don’t pay overages!

You’ll receive notifications like this:

<p align="center">
  <img src="https://github.com/mixpanel/docs/assets/35945133/6a374eb8-5be4-42b6-ad03-647226b5d911" width="50%" height="50%" />
</p>

*Forewarned is forearmed!*

### 3. Create custom alerts

If you want to go the extra mile, an additional step to detect (**even sooner!**) that your data usage has increased unexpectedly is to create custom alerts based on daily usage. For example, if your project usually tracks 100k daily events, you’ll probably want to be notified if 500k events were suddenly ingested in your project, as this would be considered an unusually large data usage.

To make these notifications possible, we’ll use the [“Alerts”](https://docs.mixpanel.com/docs/features/alerts) feature over an Insights report that checks your daily number of events/unique users:
1. Create and save an Insights report querying “All Events” with daily linear segments. You can use [this one](https://mixpanel.com/project/3018488/view/3536632/app/insights#i6e7NiANXE8u) in our E-Commerce demo data set as an example.

<p align="center">
<img alt="2026_ECommerce_demo_report" src="https://github.com/user-attachments/assets/a19c8413-ad7f-4c86-8d76-931ff339d027" />
</p>

*💬 If your plan is based on events, you’ll need to create a report with the “Total Events” calculation. However, if your plan is based on MTUs, you’ll need to create it with the "Uniques" default one.*
  
2. Once the report is saved, you’ll be able to create an alert by clicking the *Options* button at the top-right side of the screen, then “Alerts”, and then “Create Alert”.

<p align="center">
<img width="40%" height="40%" alt="2026_Create_Alerts_Option" src="https://github.com/user-attachments/assets/85a487d4-2946-44db-8830-cbb8e22f721d" />
</p>
    
3. A pop-up window appears, allowing you to customize your alert's settings, setting either a custom threshold that needs to be met for the alert to trigger, or relying on our automatic [“Anomaly detection”](https://docs.mixpanel.com/docs/features/alerts#anomaly-detection) feature (only available on the Enterprise plan). Finally, set the rest of the parameters (notification frequency, method, etc), and click “Create Alert”.

<p align="center">
<img width="60%" height="60%" alt="2026_Create_Custom_Alert_window" src="https://github.com/user-attachments/assets/8b29430c-f2bf-4109-8449-65a05ac2eccf" />
</p>

### Bonus ideas:
- If you are on an Enterprise plan, you can use the **[“Event Approval”](https://docs.mixpanel.com/docs/data-governance/event-approval)** feature to control when new unexpected event types are ingested. This can also point out new features being tracked that could affect your data usage needs.
- Check your project's Lexicon for any event that hasn’t been queried in the last 30 days - events with a high volume of ingestion but no queries are consuming your data usage without generating any analytics or value. Perhaps this is the moment to stop tracking them!

**Comment by lewis-smith** [comment]
It's no longer possible to set up an alert: "Anomaly detection is available to users on Enterprise plans, to upgrade, please contact the Mixpanel sales team.". 

Of less concern, a lot of these screenshots are quite out of date.

**Reply by anaerquicia** [reply]
Hi Lewis!

Thanks for bringing this to our attention, and apologies for the outdated information. I've made a note of it and will update this article shortly.

Best regards,
**Ana** (she/her)
_Support Engineer @ Mixpanel_
